test_conformance/extensions/cl_khr_command_buffer/command_buffer_profiling.cpp

```diff
@@ -0,0 +1,295 @@
+//
+// Copyright (c) 2022 The Khronos Group Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include "basic_command_buffer.h"
+#include "procs.h"
+
+#include <vector>
+
+namespace {
+
+////////////////////////////////////////////////////////////////////////////////
+// Command-buffer profiling test cases:
+// -all commands are recorded to a single command-queue
+// -profiling a command-buffer with simultaneous use
+
+template <bool simultaneous_request>
+struct CommandBufferProfiling : public BasicCommandBufferTest
+{
+    CommandBufferProfiling(cl_device_id device, cl_context context,
+                           cl_command_queue queue)
+        : BasicCommandBufferTest(device, context, queue), wait_event(nullptr)
+    {
+        simultaneous_use_requested = simultaneous_request;
+        if (simultaneous_request) buffer_size_multiplier = 2;
+    }
+
+    //--------------------------------------------------------------------------
+    bool Skip() override
+    {
+        if (BasicCommandBufferTest::Skip()) return true;
+
+        cl_int error = CL_SUCCESS;
+        cl_queue_properties_khr device_props = 0;
```

Comment:
The [CL_DEVICE_QUEUE_PROPERTIES](https://registry.khronos.org/OpenCL/specs/3.0-unified/html/OpenCL_API.html#CL_DEVICE_QUEUE_PROPERTIES) query returns a `cl_command_queue_properties` type not a `cl_queue_properties_khr`

Reply:
Correction applied, one remark though. In other conformance test (which was probably the sample I follow) we have the same mistake: 

[    cl_queue_properties_khr device_props = 0;](https://github.com/KhronosGroup/OpenCL-CTS/blob/b81b49e10e18ad5b059a3e3af51b209373c1c7aa/test_conformance/api/test_queue_properties.cpp#L110)
...
[    error = clGetDeviceInfo(deviceID, CL_DEVICE_QUEUE_PROPERTIES, sizeof(device_props), &device_props, NULL);](https://github.com/KhronosGroup/OpenCL-CTS/blob/b81b49e10e18ad5b059a3e3af51b209373c1c7aa/test_conformance/api/test_queue_properties.cpp#L127)

Worth raising an issue ?

Reply:
ah good to know, filed issue https://github.com/KhronosGroup/OpenCL-CTS/issues/1640 to address this